search-app/src/cmr/search/services/query_service.clj

```diff
@@ -301,6 +304,102 @@
      (ph/provider-holdings->string
        (:result-format params) provider-holdings {:echo-compatible? (= "true" echo-compatible)})]))
 
+(defn- get-collections-with-deleted-revisions
+  "Executes elasticsearch searches to find collections that have deleted revisions.
+   Returns a list of concept-ids of the collections."
+  [context params]
+  ;;1/ Find all collection revisions that are deleted satisfy the original search params
+  (let [query (make-concepts-query context :collection params)
+        condition (gc/and-conds
+                   [(:condition query)
+                    (qm/boolean-condition :deleted true)])
+        results (common-idx/execute-query context
+                                          (-> query
+                                              (assoc :all-revisions? true)
+                                              (assoc :condition condition)
+                                              (assoc :page-size :unlimited)))
+        coll-concept-ids (map #(first (:concept-id (:fields %)))
+                              (get-in results [:hits :hits]))]
+    (distinct coll-concept-ids)))
+
+(defn- get-visible-collections
+  "Returns the concept ids of collections that are visible from the given collection concept ids"
+  [context coll-concept-ids]
+  (when (seq coll-concept-ids)
+    (let [condition (qm/string-conditions :concept-id coll-concept-ids true)
```

Comment:
After reading the code a bit more, the OR string conditions are not turned into boolean or conditions, they are turned to term condition with multiple values. So there is no limit on that as the test has proven.